dokuwiki-syntax/src/main/java/org/xwiki/contrib/dokuwiki/syntax/internal/parser/DokuWikiIterativeParser.java

```diff
@@ -958,18 +958,40 @@ private void processLink(Reader source, Listener listener) throws IOException
                             new ArrayList<>(functionBuffer
                                     .subList(1, functionBuffer.size() - 2))), ResourceType.MAILTO);
                     reference.setTyped(true);
-                } else {
-                    reference = new ResourceReference(getStringRepresentation(
-                            new ArrayList<>(functionBuffer.subList(0, functionBuffer.size() - 1))), ResourceType.URL);
-                    reference.setTyped(false);
+                    listener.beginLink(reference, false, Listener.EMPTY_PARAMETERS);
+                    processWordsFromReader((char) c, source, listener, TAG_DOUBLE_CLOSING_SQUARE_BRACKETS);
+                    listener.endLink(reference, false, Listener.EMPTY_PARAMETERS);
+                    break;
+                }
+                else {
+                    if (c != '{'){
+                        reference = new ResourceReference(getStringRepresentation(
+                                new ArrayList<>(functionBuffer.subList(0, functionBuffer.size() - 1))), ResourceType.URL);
+                        reference.setTyped(false);
+                        listener.beginLink(reference, false, Listener.EMPTY_PARAMETERS);
+                        functionBuffer.add((char) c);
+                        processWordsFromReader((char) c, source, listener, TAG_DOUBLE_CLOSING_SQUARE_BRACKETS);
+                        listener.endLink(reference, false, Listener.EMPTY_PARAMETERS);
+                        break;
+                    } else {
+                        functionBuffer.add((char) c);
+                    }
                 }
+            }
+
+            if (getStringRepresentation(functionBuffer).endsWith("{{")) {
+                reference = new ResourceReference(getStringRepresentation(new ArrayList<>(functionBuffer.subList(0, functionBuffer.size() - 3))), ResourceType.URL);
```

Comment:
You should not assume it's always going to be a URL type here, as far as I can see in your tests it's really not the case.

Reply:
I'm not sure about this. I found that according to DokuWiki syntax anything that enters this loop will definitely be atleast an URL. The image syntax processed after can be an attachment.

Reply:
I don't think you understand what the type refer to. If your reference is not of the form "protocol://some/path" it's not an URL so an attachment is not an URL, you have an attachment type for those.